.github/workflows/docs.yml

```diff
@@ -0,0 +1,40 @@
+name: Build docs
+
+on:
+  push:
+    branches: [master]
+    tags: ["v*"]
+  release:
+    types: [created]
+  pull_request:
+    branches: [master]
+
+jobs:
+  build_docs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: "3.7"
+      - name: install docs reqs
+        shell: bash -l {0}
+        run: |
+          pip install -U pip wheel
+          pip install torch torchvision -f https://download.pytorch.org/whl/cpu/torch_stable.html -U
+          pip install -r requirements-dev.txt
+          pip install -r docs/requirements.txt
+          pip install git+https://github.com/vfdev-5/sphinxcontrib-versioning.git
+
+      - name: build docs
+        shell: bash -l {0}
+        run: sphinx-versioning --use-master-conf --use-master-templates build --greatest-tag --whitelist-branches master docs/source docs/build/html
+
+      # - name: deploy docs
```

Comment:
Deploy part should be uncommented and have if condition on pushed to master or on tag, right ?

Reply:
There is `on: push - branches: [master]` and `tag: ["v*"]`. Yes, it should be uncommented, but no need for if condition, I think. Can you verify `tag` will work as I haven't used `tag` before?

Reply:
Yes, you are right about push and tag. Probably, should work like that (to verfiy in the playground). My point is about pull request. Actually, I forgot that we wanted to build docs with netlify, so we can remove `pull_request:
branches: [master]` from here, I think. Thus, we build and deploy on push-master and on tag. PR's preview is done on netlify and it was also a nice to have from https://github.com/pytorch/ignite/issues/1270#issuecomment-703180980 and the idea is to refactor deps installation and docs building into 2 scripts that could be used here and in netlify. Could you please create those 2 scripts inside `.github/workflows` and reuse them here ?

Reply:
I see!
That line should be uncommented, just want to test via PR if docs can be built and yes `pull_request` needs to be removed.

Reply:
@vfdev-5 when I tested on my fork, `tags` works if you do with `git push --tags` and make a tag that starts with `v` from github UI. `release` only works if you make a release/tag from github UI.

Reply:
Both are OK, I think